src/python/impactx/dashboard/Input/space_charge_configuration/spaceChargeMain.py

```diff
@@ -0,0 +1,328 @@
+from trame.widgets import vuetify
+
+from ...trame_setup import setup_server
+from ..generalFunctions import generalFunctions
+from .spaceChargeFunctions import SpaceChargeFunctions
+
+server, state, ctrl = setup_server()
+
+# -----------------------------------------------------------------------------
+# Default
+# -----------------------------------------------------------------------------
+
+state.dynamic_size = False
+state.max_level = 0
+state.n_cell = []
+state.prob_relative = []
+state.particle_shape = 2
+state.poisson_solver = "fft"
+
+state.prob_relative_fields = []
+state.n_cell_x = 32
+state.n_cell_y = 32
+state.n_cell_z = 32
+
+state.blocking_factor_x = 16
+state.blocking_factor_y = 16
+state.blocking_factor_z = 16
+
+state.mlmg_relative_tolerance = 1.0e-7
+state.mlmg_absolute_tolerance = 0.0
+state.mlmg_max_iters = 100
+state.mlmg_verbosity = 1
+
+state.error_message_mlmg_relative_tolerance = ""
+state.error_message_mlmg_absolute_tolerance = ""
+state.error_message_mlmg_max_iters = ""
+state.error_message_mlmg_verbosity = ""
+
+# -----------------------------------------------------------------------------
+# Helper functions
+# -----------------------------------------------------------------------------
+
+
+def populate_prob_relative_fields(max_level):
+    num_prob_relative_fields = int(max_level) + 1
+
+    if state.poisson_solver == "fft":
+        state.prob_relative = [1.1] + [0.0] * (num_prob_relative_fields - 1)
+    elif state.poisson_solver == "multigrid":
+        state.prob_relative = [3.1] + [0.0] * (num_prob_relative_fields - 1)
+    else:
+        state.prob_relative = [0.0] * num_prob_relative_fields

```

Comment:
Question for both you and @cemitch99, should we add a comment about the numbers that appear here (e.g., `1.1`, `3.1`, etc.)? I'm not too familiar with the context here, so I'm not sure if the meaning of these numbers should be clarified.

Reply:
Those numbers in `prob_relative` specify the size of the grid relative to the size of the beam, and are documented [here](https://impactx.readthedocs.io/en/latest/usage/parameters.html#collective-effects) and [here](https://impactx.readthedocs.io/en/latest/usage/python.html#collective-effects-overall-simulation-parameters).  Maybe some of the input parameters should include links to the documentation?